include/zephyr/bluetooth/conn.h

```diff
@@ -413,6 +413,130 @@ struct bt_conn_le_cs_fae_table {
 	uint8_t *remote_fae_table;
 };
 
+/** Channel sounding main mode */
+enum bt_conn_le_cs_main_mode {
```

Comment:
There seems to be some inconsistency with the API naming. For the public functions you only use "cs", e.g. "bt_cs_create_config()", however here you use "le_cs". I think it would be good either use the "le_" prefix everywhere or nowhere (my preference is the latter). The only exception could be the HCI-related definitions, since there it's better to follow whatever the HCI spec calls the events, commands and associated parameters.

Reply:
You are right. API exposed from `cs.h` is specific to LE CS so I agree with you that we don't need to name it `le_cs` apart from the HCI parts.

Where the inconsistency happens is in `conn.h` which appear to contain both LE and bluetooth classic features. Additional prefix there might be useful to distinguish between the two but I see your point in terms of the inconsistency.

Happy to change the cs related definitions in `conn.h` if you prefer that. One downside to that will be the inconsistency of the naming between some other LE features (`bt_conn_le_subrating*`, `bt_conn_le_tx_power*` etc.) and CS.

Reply:
I took a look at existing LE-related definitions in `conn.h` and it seems the convention is to have `le_` here, so it's probably best to just follow this. What's also messy, is that the `bt_conn_cb` members (even non-CS related ones) *don't* have `le_` even though their other corresponding APIs do have it. (e.g. LE power reporting or LE Subrating related APIs).

Reply:
Good point with the callbacks. Not sure about the previous features but I lean towards adding le for CS callbacks at least. Just so we can stick to a standard for the newly added codes. In long term we would have more callbacks that follow the correct approach this way.

What do you think?

Reply:
@jori-nordic @alwa-nordic @carlescufi any thoughts regarding the conventions & consistency with LE-specific Bluetooth APIs?

Reply:
The Zephyr Bluetooth host stack has supported both BR/EDR and LE since the beginning (2015 or so) and the approach has been that anything LE-specific should have "le" in the name. Since then, especially when it seemed like no-one was using BR/EDR and it might be completely removed, there's been a trend to simplify things and just omit the "LE" part. Now that Bluetooth Classic support has been revived it seems like this might not have been that wise after all.

Reply:
Since classic is on its way out, I would err on the side of removing `le_` and instead adding `br_` on APIs.

Reply:
> Since classic is on its way out, I would err on the side of removing `le_` and instead adding `br_` on APIs.

No objections to this, but to be clear it's only "on the way out" from a Bluetooth SIG long-term plan perspective. In a Zephyr context I'd say it's very much "on the way in", at least based on the ever growing list of pull requests I have to review :)

Reply:
@lylezhu2012 FYI (see earlier comments)

Reply:
haha yes for sure. Any way, having a clear split would be good for new (both bluetooth & zephyr) users.

Reply:
+1 for `bt_le_` for clear division.

Reply:
Will add `le` prefix for CS callbacks if there are no objections then

Reply:
> Will add `le` prefix for CS callbacks if there are no objections then

Hmm, @jori-nordic said that he'd prefer removing `le_`, to which I said I have no objections. @alwa-nordic said the opposite. I don't think anybody was strongly objecting one way or another, but I didn't get the impression that there was a strong consensus either. What I care about is that we try to move towards API consistency, especially within specific features (like CS) but also within the Bluetooth API in general. As for BR/EDR, I *think* we should be in pretty good shape there, i.e. most of those APIs should already have `br_` in the names.

Reply:
> Any way, having a clear split would be good for new (both bluetooth & zephyr) users.

I interpreted this as clear split with the use of `le` and `br` prefixes but maybe I misunderstood.

To move this forward, do you suggest applying changes, leaving as is or having further discussion?

Reply:
Seeing the amount of `le_` around in the API, I would prefer `bt_le_` and `bt_br_`.

Reply:
Ok. Let's go with that then, so please make sure any public functions have either `bt_conn_le_cs_*` or `bt_le_cs_*` as their name and any `bt_conn_cb` callbacks `le_cs_*`.